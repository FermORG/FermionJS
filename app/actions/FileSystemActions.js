// @flow
export const SET_ACTIVE_COMPONENT = 'SET_ACTIVE_COMPONENT';

export function setActiveComponent(component: string) {
  return {
    type: SET_ACTIVE_COMPONENT,
    component,
  };
}
