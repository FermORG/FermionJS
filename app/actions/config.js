// @flow
// config actions
export const ADD_STATE = 'ADD_STATE';
export const ADD_PROPS = 'ADD_PROPS';
export const ADD_STYLES = 'ADD_STYLES';
export const ADD_EVENTS = 'ADD_EVENTS';
export const DELETE_STATE = 'DELETE_STATE';
export const DELETE_PROPS = 'DELETE_PROPS';
export const DELETE_STYLES = 'DELETE_STYLES';
export const DELETE_EVENTS = 'DELETE_EVENTS';

export function addState(aState, component) {
  return {
    type: 'ADD_STATE',
    aState,
  };
}

export function addProps(prop, component) {
  return {
    type: 'ADD_PROPS',
    prop,
    component
  };
}

export function addStyles(style, component) {
  return {
    type: 'ADD_STYLES',
    style,
    component
  };
}

export function addEvents(event, component) {
  return {
    type: 'ADD_EVENTS',
    event,
    component
  };
}

export function deleteState(stateKey, component) {
  return {
    type: 'DELETE_STATE',
    propKey: stateKey,
  };
}
export function deleteProps(propKey, component) {
  return {
    type: 'DELETE_PROPS',
    prop: propKey,
    component
  };
}
export function deleteStyles(styleKey, component) {
  return {
    type: 'DELETE_STYLES',
    style: styleKey,
    component
  };
}
export function deleteEvents(eventKey, component) {
  return {
    type: 'DELETE_EVENTS',
    event: eventKey,
    component
  };
}
