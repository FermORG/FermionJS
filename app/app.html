<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <title>Fermion</title>
    <script>
      (function() {
        if (!process.env.HOT) {
          const link = document.createElement('link');
          link.rel = 'stylesheet';
          link.href = './dist/style.css';
          // HACK: Writing the script path should be done with webpack
          document.getElementsByTagName('head')[0].appendChild(link);
        }
      }());
    </script>
<link href="https://fonts.googleapis.com/css?family=Concert+One|Hind|Inconsolata|Josefin+Slab|Julius+Sans+One|Lato|Meera+Inimai|Noto+Sans|Open+Sans|Poiret+One|Raleway|Roboto|Saira+Semi+Condensed|Ubuntu|Ubuntu+Condensed|Bungee" rel="stylesheet">
  </head>
  <body>
    <div id="root"></div>
    <script>
      {
        const scripts = [];

        // Dynamically insert the DLL script in development env in the
        // renderer process
        if (process.env.NODE_ENV === 'development') {
          scripts.push('../dll/renderer.dev.dll.js');
        }

        // Dynamically insert the bundled app script in the renderer process
        const port = process.env.PORT || 1212;
        scripts.push(
          (process.env.HOT)
            ? 'http://localhost:' + port + '/dist/renderer.dev.js'
            : './dist/renderer.prod.js'
        );

        document.write(
          scripts
            .map(script => '<script defer src="' + script + '"><\/script>')
            .join('')
        );
      }
    </script>
  </body>
</html>
