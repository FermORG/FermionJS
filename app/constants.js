export default {
  WORKSPACE_ID: 'workspace'
};