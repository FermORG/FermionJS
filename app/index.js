import React from 'react';
import { render } from 'react-dom';
import { AppContainer } from 'react-hot-loader';
import Root from './containers/Root';
import { configureStore, history } from './store/configureStore';
import './app.global.css';
import { ipcRenderer } from 'electron';

ipcRenderer.on('ConfigApp', (event, data) => {
  data = JSON.parse(data);

  const initialState = {
    availableComponents: data.availableComponents,
  }
 const store = configureStore(initialState);

  render(
    <AppContainer>
      <Root store={store} history={history} />
    </AppContainer>,
    document.getElementById('root')
  );

  if (module.hot) {
    module.hot.accept('./containers/Root', () => {
      const NextRoot = require('./containers/Root'); // eslint-disable-line global-require
      render(
        <AppContainer>
          <NextRoot store={store} history={history} />
        </AppContainer>,
        document.getElementById('root')
      );
    });
  }

});
