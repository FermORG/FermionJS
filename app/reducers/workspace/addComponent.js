import cloneComponentAndChildren from './utilities';

/**
 * Creates a new child and adds it to the provided parent component (parentID)
 * @param {object} state - workspace data
 * @param {object} action
 * -> @property {number} action.targetID - id of parent to append new component to
 * -> @property {object} action.newComponent - new component to append to target
 */
export default function addComponent(state, action) {
  const components = { ...state.components };
  components.workspace = cloneComponentAndChildren(components.workspace);
  const { newComponent } = action;
  newComponent.id = state.componentCounter;
  components[newComponent.id] = newComponent;
  components.workspace.children.push(state.componentCounter);

  return {
    ...state,
    componentCounter: state.componentCounter + 1,
    components,
  };
}
