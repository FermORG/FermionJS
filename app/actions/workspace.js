import { WORKSPACE_ID } from './../constants';

export const ADD_CHILD = 'ADD_CHILD';
export const CREATE_CHILD = 'CREATE_CHILD';
export const DELETE_CHILD = 'DELETE_CHILD';
export const MOVE_CHILD = 'MOVE_CHILD';
export const REMOVE_CHILD = 'REMOVE_CHILD';
export const UPDATE_STYLE = 'UPDATE_STYLE';

const createChild = ({ name, props }) => ({
  name,
  props,
  children: [],
  parentID: WORKSPACE_ID,
});

// export function addChild(targetID, newComponentName) {
//   const newComponent = createChild(newComponentName);
export function addChild(targetID, component) {
  const newComponent = createChild(component);
  return {
    type: ADD_CHILD,
    newComponent,
    targetID
  };
}

export function removeChild(id) {
  return {
    type: REMOVE_CHILD,
    id
  };
}

export function moveChild(sourceID, targetID) {
  return {
    type: MOVE_CHILD,
    targetID,
    sourceID
  };
}

export function updateStyle(sourceID, newStyle) {
  return {
    type: UPDATE_STYLE,
    sourceID,
    newStyle
  };
}
