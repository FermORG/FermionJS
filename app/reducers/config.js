// @flow
// config reducer

import { ADD_STATE, ADD_PROPS, ADD_STYLES, ADD_EVENTS, CHANGE_STATE, CHANGE_PROPS, CHANGE_STYLES, CHANGE_EVENTS } from '../actions/config';
import { WORKSPACE_ID } from './../constants';

export type configStateType = {
  // what goes here?
};

type actionType = {
  +type: string
};

const addStateValue = (state, aStateValue) => {
  const newState = Object.assign({}, state);
  const newAppState = Object.assign({}, newState.state, aStateValue);
  newState.state = newAppState;
  return newState;
};


export default function toggleConfig(state: configStateType = {}, action: actionType) {
  switch (action.type) {
    case ADD_STATE:
      return addStateValue(state, action.aState);

    case ADD_PROPS:
      return state;

    case ADD_STYLES:
      return state;

    case ADD_EVENTS:
      return state;

    case CHANGE_STATE:
      return state;

    case CHANGE_PROPS:
      return state;

    case CHANGE_STYLES:
      return state;

    case CHANGE_EVENTS:
      return state;

    default:
      return state;
  };
}
