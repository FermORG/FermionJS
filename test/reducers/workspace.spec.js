import { ADD_CHILD, REMOVE_CHILD, MOVE_CHILD, DELETE_CHILD, UPDATE_STYLE } from '../../app/actions/workspace';
import { ADD_STATE, ADD_PROPS, ADD_STYLES, ADD_EVENTS, DELETE_STATE, DELETE_PROPS, DELETE_STYLES, DELETE_EVENTS } from '../../app/actions/config';
import { SET_ACTIVE_COMPONENT } from '../../app/actions/FileSystemActions';
import { WORKSPACE_ID } from '../../app/constants';
import addComponent from '../../app/reducers/workspace/addComponent';
import removeComponent from '../../app/reducers/workspace/removeComponent';
import moveChild from '../../app/reducers/workspace/moveChild';
import addStateValue from '../../app/reducers/workspace/addStateValue';
import addPropsValue from '../../app/reducers/workspace/addPropsValue';
import addStyleValue from '../../app/reducers/workspace/addStyleValue';
import addEvent from '../../app/reducers/workspace/addEvent';
import deleteStateValue from '../../app/reducers/workspace/deleteStateValue';
import deletePropsValue from '../../app/reducers/workspace/deletePropsValue';
import deleteStylesValue from '../../app/reducers/workspace/deleteStylesValue';
import deleteEvent from '../../app/reducers/workspace/deleteEvent';
import setActiveComponent from '../../app/reducers/workspace/setActiveComponent';
import updateStyle from '../../app/reducers/workspace/updateStyle';

import workspace from '../../app/reducers/workspace/index';

const defaultWorkspace = {
  componentCounter: 2,
  activeComponent: '0',
  components: {
    workspace: {
      id: WORKSPACE_ID,
      children: [0, 1],
    },
  },
  state: {},
};

defaultWorkspace.components[0] = {
  id: 0,
  name: 'BlackBox',
  children: [],
  parentID: WORKSPACE_ID,
  props: {
    style: {
      position: 'relative',
      height: '100px',
      width: '100px',
      display: 'inline-block',
      backgroundColor: 'black',
      overflow: 'auto',
    },
  },
  events: { test: 'test' }
};

defaultWorkspace.components[1] = {
  id: 1,
  name: 'BlueBox',
  children: [],
  parentID: WORKSPACE_ID,
  props: {
    style: {
      position: 'relative',
      height: '100px',
      width: '100px',
      display: 'inline-block',
      backgroundColor: 'blue',
      overflow: 'auto',
    },
  },
  events: { test: 'event test =]' }
};



describe('workspace Reducer', () => {
  describe('CASE: ADD CHILD', () => {
    it('should add a child to the workspace state', () => {
      const actionMock = {
        type: ADD_CHILD,
        newComponent: {
          id: 2,
          name: 'test',
          props: {
            style: {
              position: 'relative',
            },
          },
          events: {test: 'test'}
        },
      };
      const result = workspace(defaultWorkspace, actionMock);
      expect(result).toMatchSnapshot();
      expect(result).toHaveProperty('componentCounter', 3);
      expect(result).toHaveProperty('components.workspace.children', [0,1,2]);
        //write some more here later, this is a big object and needs to be controlled.
    });
  });
  // too big for tonight
  // describe('CASE: REMOVE CHILD', () => {
  //   it('should remove a child from the workspace state', () => {
  //     const actionMock = {
  //       type: REMOVE_CHILD,
  //     }
  //   });
  // });

  describe('CASE: MOVE_CHILD', () => {
    it('should nest a component in another component', () => {
      const actionMock = {
        type: MOVE_CHILD,
        targetID: 1,
        sourceID: 0,
      };
      const result = workspace(defaultWorkspace, actionMock);
      expect(result).toMatchSnapshot();
      expect(result).toHaveProperty('componentCounter', 2);
      expect(result.components[1]).toHaveProperty('children', [0]);
    });

    it('should not move a component into itself', () => {
      const actionMock = {
        type: MOVE_CHILD,
        targetID: 1,
        sourceID: 1,
      };

      const result = workspace(defaultWorkspace, actionMock);
      expect(result).toMatchObject(defaultWorkspace);
    });

    it('should not move a component into its own parent', () => {
      const actionMock = {
        type: MOVE_CHILD,
        targetID: WORKSPACE_ID,
        sourceID: 0,
      };

      const result = workspace(defaultWorkspace, actionMock);
      expect(result).toMatchObject(defaultWorkspace);
    })
  });

  describe('CASE: ADD STATE', () => {
    it('should add a value to the workspace state object', () => {
      const actionMock = {
        type: ADD_STATE,
        aState: {
          'test': null,
        },
      };

      const result = workspace(defaultWorkspace, actionMock);
      expect(result).toMatchSnapshot();
      expect(result).toHaveProperty('state.test', null);
    });
  });

  describe('CASE: ADD_PROPS', () => {
    it('should add a prop to a component', () => {
      const actionMock = {
        type: ADD_PROPS,
        prop: {
          test: null,
        },
        component: 0,
      };

      const result = workspace(defaultWorkspace, actionMock);
      expect(result).toMatchSnapshot();
      expect(result.components[0]).toHaveProperty('props.test', null);
    });

    it('should not add a style prop', () => {
      const actionMock = {
        type: ADD_PROPS,
        prop: {
          style: null,
        },
        component: 0,
      };
      const result = workspace(defaultWorkspace, actionMock);
      expect(result).toMatchObject(defaultWorkspace);
    })
  });

  describe('CASE: ADD_STYLES', () => {
    it('should add a new style to props.style', () => {
      const actionMock = {
        type: ADD_STYLES,
        style: { border: null },
        component: 0,
      };

      const result = workspace(defaultWorkspace, actionMock);

      expect(result).toMatchSnapshot();
      expect(result.components[0].props.style).toHaveProperty('border', null);
    });
  });

  describe('CASE: ADD_EVENTS', () => {
    it('should add a new event to a component', () => {
      const actionMock = {
        type: ADD_EVENTS,
        event: { onClick: null },
        component: 0,
      };

      const result = workspace(defaultWorkspace, actionMock);

      expect(result).toMatchSnapshot();
      expect(result.components[0].events).toHaveProperty('onClick', null);
    })
  })






});
