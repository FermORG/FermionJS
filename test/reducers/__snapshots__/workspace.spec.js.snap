// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workspace Reducer CASE: ADD CHILD should add a child to the workspace state 1`] = `
Object {
  "activeComponent": "0",
  "componentCounter": 3,
  "components": Object {
    "0": Object {
      "children": Array [],
      "events": Object {
        "test": "test",
      },
      "id": 0,
      "name": "BlackBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "black",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
        "testProp": null,
      },
    },
    "1": Object {
      "children": Array [],
      "events": Object {
        "test": "event test =]",
      },
      "id": 1,
      "name": "BlueBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "blue",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
      },
    },
    "2": Object {
      "events": Object {
        "test": "test",
      },
      "id": 2,
      "name": "test",
      "props": Object {
        "style": Object {
          "position": "relative",
        },
      },
    },
    "workspace": Object {
      "children": Array [
        0,
        1,
        2,
      ],
      "id": "workspace",
    },
  },
  "state": Object {
    "default": "default",
  },
}
`;

exports[`workspace Reducer CASE: ADD STATE should add a value to the workspace state object 1`] = `
Object {
  "activeComponent": "0",
  "componentCounter": 2,
  "components": Object {
    "0": Object {
      "children": Array [],
      "events": Object {
        "test": "test",
      },
      "id": 0,
      "name": "BlackBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "black",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
        "testProp": null,
      },
    },
    "1": Object {
      "children": Array [],
      "events": Object {
        "test": "event test =]",
      },
      "id": 1,
      "name": "BlueBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "blue",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
      },
    },
    "workspace": Object {
      "children": Array [
        0,
        1,
      ],
      "id": "workspace",
    },
  },
  "state": Object {
    "default": "default",
    "test": null,
  },
}
`;

exports[`workspace Reducer CASE: ADD_EVENTS should add a new event to a component 1`] = `
Object {
  "activeComponent": "0",
  "componentCounter": 2,
  "components": Object {
    "0": Object {
      "children": Array [],
      "events": Object {
        "onClick": null,
        "test": "test",
      },
      "id": 0,
      "name": "BlackBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "black",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
        "test": null,
        "testProp": null,
      },
    },
    "1": Object {
      "children": Array [],
      "events": Object {
        "test": "event test =]",
      },
      "id": 1,
      "name": "BlueBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "blue",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
      },
    },
    "workspace": Object {
      "children": Array [
        0,
        1,
      ],
      "id": "workspace",
    },
  },
  "state": Object {
    "default": "default",
  },
}
`;

exports[`workspace Reducer CASE: ADD_PROPS should add a prop to a component 1`] = `
Object {
  "activeComponent": "0",
  "componentCounter": 2,
  "components": Object {
    "0": Object {
      "children": Array [],
      "events": Object {
        "test": "test",
      },
      "id": 0,
      "name": "BlackBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "black",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
        "test": null,
        "testProp": null,
      },
    },
    "1": Object {
      "children": Array [],
      "events": Object {
        "test": "event test =]",
      },
      "id": 1,
      "name": "BlueBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "blue",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
      },
    },
    "workspace": Object {
      "children": Array [
        0,
        1,
      ],
      "id": "workspace",
    },
  },
  "state": Object {
    "default": "default",
  },
}
`;

exports[`workspace Reducer CASE: ADD_STYLES should add a new style to props.style 1`] = `
Object {
  "activeComponent": "0",
  "componentCounter": 2,
  "components": Object {
    "0": Object {
      "children": Array [],
      "events": Object {
        "test": "test",
      },
      "id": 0,
      "name": "BlackBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "black",
          "border": null,
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
        "test": null,
        "testProp": null,
      },
    },
    "1": Object {
      "children": Array [],
      "events": Object {
        "test": "event test =]",
      },
      "id": 1,
      "name": "BlueBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "blue",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
      },
    },
    "workspace": Object {
      "children": Array [
        0,
        1,
      ],
      "id": "workspace",
    },
  },
  "state": Object {
    "default": "default",
  },
}
`;

exports[`workspace Reducer CASE: DELETE STYLE should remove a key from the style object 1`] = `
Object {
  "activeComponent": "0",
  "componentCounter": 2,
  "components": Object {
    "0": Object {
      "children": Array [],
      "events": Object {
        "onClick": null,
        "test": "test",
      },
      "id": 0,
      "name": "BlackBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
        "test": null,
      },
    },
    "1": Object {
      "children": Array [],
      "events": Object {
        "test": "event test =]",
      },
      "id": 1,
      "name": "BlueBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "blue",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
      },
    },
    "workspace": Object {
      "children": Array [
        0,
        1,
      ],
      "id": "workspace",
    },
  },
  "state": Object {
    "default": "default",
  },
}
`;

exports[`workspace Reducer CASE: DELETE_PROPS should remove a value from a component's props 1`] = `
Object {
  "activeComponent": "0",
  "componentCounter": 2,
  "components": Object {
    "0": Object {
      "children": Array [],
      "events": Object {
        "onClick": null,
        "test": "test",
      },
      "id": 0,
      "name": "BlackBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "black",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
        "test": null,
      },
    },
    "1": Object {
      "children": Array [],
      "events": Object {
        "test": "event test =]",
      },
      "id": 1,
      "name": "BlueBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "blue",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
      },
    },
    "workspace": Object {
      "children": Array [
        0,
        1,
      ],
      "id": "workspace",
    },
  },
  "state": Object {
    "default": "default",
  },
}
`;

exports[`workspace Reducer CASE: DELETE_STATE should remove a value from the app state 1`] = `
Object {
  "activeComponent": "0",
  "componentCounter": 2,
  "components": Object {
    "0": Object {
      "children": Array [],
      "events": Object {
        "onClick": null,
        "test": "test",
      },
      "id": 0,
      "name": "BlackBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "black",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
        "test": null,
        "testProp": null,
      },
    },
    "1": Object {
      "children": Array [],
      "events": Object {
        "test": "event test =]",
      },
      "id": 1,
      "name": "BlueBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "blue",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
      },
    },
    "workspace": Object {
      "children": Array [
        0,
        1,
      ],
      "id": "workspace",
    },
  },
  "state": Object {},
}
`;

exports[`workspace Reducer CASE: MOVE_CHILD should nest a component in another component 1`] = `
Object {
  "activeComponent": "0",
  "componentCounter": 2,
  "components": Object {
    "0": Object {
      "children": Array [],
      "events": Object {
        "test": "test",
      },
      "id": 0,
      "name": "BlackBox",
      "parentID": 1,
      "props": Object {
        "style": Object {
          "backgroundColor": "black",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
        "testProp": null,
      },
    },
    "1": Object {
      "children": Array [
        0,
      ],
      "events": Object {
        "test": "event test =]",
      },
      "id": 1,
      "name": "BlueBox",
      "parentID": "workspace",
      "props": Object {
        "style": Object {
          "backgroundColor": "blue",
          "display": "inline-block",
          "height": "100px",
          "overflow": "auto",
          "position": "relative",
          "width": "100px",
        },
      },
    },
    "workspace": Object {
      "children": Array [
        1,
      ],
      "id": "workspace",
    },
  },
  "state": Object {
    "default": "default",
  },
}
`;
